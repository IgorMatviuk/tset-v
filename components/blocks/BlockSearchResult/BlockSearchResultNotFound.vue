<template>
  <div class="no-search-result">
    <BContainer>
      <SearchBox
        :with-label="true"
        :initial-value="query"
        class="no-search-result__box"
      />
      <div class="no-search-result__wrapper">
        <svg-icon
          name="dress"
          class="no-search-result__icon"
          width="107"
          height="107"
        />
        <div class="no-search-result__title">
          {{ $t('searchPage.notFoundTitle') }}
        </div>
        <div class="no-search-result__description">
          {{ $t('searchPage.notFoundDescription') }}
        </div>
        <BButton
          variant="primary"
          :to="backToShoppingLink"
          class="no-search-result__to-catalog"
        >
          {{ $t('buttons.catalog') }}
        </BButton>
      </div>
    </BContainer>
  </div>
</template>

<script>
import SearchBox from '@/components/functional/SearchBox'

export default {
  components: {
    SearchBox
  },
  props: {
    query: {
      type: String,
      required: true
    }
  },
  computed: {
    backToShoppingLink() {
      return this.$i18n.path(this.$store.getters['catalog/defaultCategorySlug'])
    }
  }
}
</script>

<style lang="scss">
.no-search-result {
  min-height: calc(100vh - 430px);
  padding-top: 8px;
  padding-bottom: 50px;

  &__box {
    max-width: 690px;
    margin-right: auto;
    margin-left: auto;

    .search-box {
      &__label {
        display: block;
        margin-bottom: 20px;
        font-size: 16px;
        text-align: center;
      }
    }
  }

  &__wrapper {
    padding-top: 57px;
    text-align: center;
  }

  &__icon {
    width: 107px;
    height: 107px;
    margin-bottom: 24px;
    color: $primary-light-5;
  }

  &__title {
    max-width: 535px;
    margin-right: auto;
    margin-bottom: 18px;
    margin-left: auto;
    font-size: 16px;
    font-weight: 700;
    line-height: 20px;
    letter-spacing: 0.02em;
  }

  &__description {
    max-width: 535px;
    margin-right: auto;
    margin-bottom: 31px;
    margin-left: auto;
    font-size: 13px;
    line-height: 16px;
    color: $gray-dark-2;
  }

  @include media-breakpoint-down(xl) {
    &__box {
      max-width: 570px;
    }

    &__wrapper {
      padding-top: 50px;
    }

    &__icon {
      width: 90px;
      height: 90px;
      margin-bottom: 22px;
    }

    &__description {
      margin-bottom: 25px;
    }
  }

  @include media-breakpoint-down(lg) {
    min-height: calc(100vh - 480px);

    &__box {
      max-width: 450px;

      .search-box {
        &__label {
          margin-bottom: 15px;
        }
      }
    }

    &__wrapper {
      padding-top: 40px;
    }

    &__icon {
      width: 80px;
      height: 80px;
      margin-bottom: 20px;
    }

    &__title {
      max-width: 450px;
      margin-bottom: 10px;
      font-size: 14px;
      line-height: 18px;
    }

    &__description {
      max-width: 450px;
      margin-bottom: 20px;
      font-size: 12px;
      line-height: 15px;
    }
  }

  @include media-breakpoint-down(md) {
    min-height: calc(100vh - 590px);

    &__box {
      .search-box {
        &__lable {
          margin-bottom: 18px;
        }
      }
    }

    &__wrapper {
      padding-top: 38px;
    }

    &__icon {
      display: none;
    }

    &__title {
      margin-bottom: 10px;
    }

    &__descriptions {
      margin-bottom: 20px;
    }
  }

  @include media-breakpoint-down(xs) {
    min-height: calc(100vh - 510px);
    padding-top: 25px;
    padding-bottom: 70px;

    &__box {
      max-width: 310px;

      .search-box {
        &__label {
          margin-bottom: 19px;
        }
      }
    }

    &__wrapper {
      padding-top: 40px;
    }

    &__title {
      margin-bottom: 15px;
    }

    &__description {
      margin-bottom: 25px;
    }
  }
}
</style>
