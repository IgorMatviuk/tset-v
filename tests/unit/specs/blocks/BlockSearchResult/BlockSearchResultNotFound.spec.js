import { shallowMount, createLocalVue } from '@vue/test-utils'
import { BContainer, BButton } from 'bootstrap-vue'

import BlockSearchResultNotFound from '@/components/blocks/BlockSearchResult/BlockSearchResultNotFound'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)
localVue.component('BButton', BButton)

describe('BlockSearchResultNotFound', () => {
  let propsData

  beforeEach(() => {
    propsData = {
      query: 'Платья'
    }
  })

  test('has expected structure', () => {
    const wrapper = shallowMount(BlockSearchResultNotFound, {
      localVue,
      propsData,
      stubs: ['svg-icon'],
      mocks: {
        $t: (str) => str,
        $i18n: {
          path: (slug) => slug
        },
        $store: {
          getters: {
            'catalog/defaultCategorySlug': 'default-category-slug'
          }
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
