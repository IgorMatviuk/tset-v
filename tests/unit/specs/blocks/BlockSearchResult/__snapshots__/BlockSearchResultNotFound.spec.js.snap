// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockSearchResultNotFound has expected structure 1`] = `
<div
  class="no-search-result"
>
  <bcontainer-stub
    tag="div"
  >
    <searchbox-stub
      class="no-search-result__box"
      initialvalue="Платья"
      withlabel="true"
    />
     
    <div
      class="no-search-result__wrapper"
    >
      <svg-icon-stub
        class="no-search-result__icon"
        height="107"
        name="dress"
        width="107"
      />
       
      <div
        class="no-search-result__title"
      >
        
        searchPage.notFoundTitle
      
      </div>
       
      <div
        class="no-search-result__description"
      >
        
        searchPage.notFoundDescription
      
      </div>
       
      <bbutton-stub
        class="no-search-result__to-catalog"
        tag="button"
        target="_self"
        to="default-category-slug"
        type="button"
        variant="primary"
      >
        
        buttons.catalog
      
      </bbutton-stub>
    </div>
  </bcontainer-stub>
</div>
`;
